src/kvstore/raftex/RaftPart.cpp

```diff
@@ -977,10 +986,15 @@ void RaftPart::statusPolling() {
         VLOG(2) << idStr_ << "Need to send heartbeat";
         sendHeartbeat();
     }
-    wal_->cleanWAL();
+    if (needToCleanupSnapshot()) {
```

Comment:
maybe else if is better

Reply:
We can't use 'else if'.   Because as follower,  check leader election and check snapshot  are two actions independently.

Reply:
and I see the code that in function "RaftPart::needToSendHeartbeat" and "RaftPart::needToStartElection" it check condition "status_ == Status::RUNNING", it is different status with "status_ == Status::WAITING_SNAPSHOT", and this function never change the status, so I think need use else if.

and more, the state machine of raft is too complex now, is there some method to make it easy?

Reply:
Yes.  Currently, the logic is the same with using "else if".  But they are two independent actions.